Explorer/Assets/DCL/Landscape/Worlds/TerrainModel.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Mathematics;
+using UnityEngine;
+
+namespace DCL.Landscape
+{
+    public class TerrainModel
+    {
+        // Note: [units] = Unity units (1 unit = 1 meter)
+
+        private const int MAX_CHUNK_SIZE = 512; // Maximum size of a chunk in Unity [units]
+        private const int MIN_CHUNKS_PER_SIDE = 2; // Minimum number of chunks along one side of terrain, ensuring at least a 2x2 grid
+
+        private readonly int parcelSize; // Size of a [parcel] in Unity [units]
+
+        public readonly int2 MinParcel;
+        public readonly int2 MaxParcel;
+
+        public readonly int2 SizeInUnits;
+        public readonly int2 MinInUnits;
+        public readonly int2 MaxInUnits;
+
+        public int ChunkSizeInUnits; //  in [units]
+
+        public List<ChunkModel> ChunkModels;
+
+        private int chunkSizeInParcels; //  in [parcels]
+        private int sizeInChunks; // Number of chunks along one side of the square terrain
+
+        public TerrainModel(int parcelSize, WorldModel world, int paddingInParcels)
+        {
+            this.parcelSize = parcelSize;
+
+            ChunkSizeInUnits = 0;
+            chunkSizeInParcels = 0;
+            sizeInChunks = 0;
+
+            int2 sizeInParcels = world.SizeInParcels + (2 * paddingInParcels);
+            int2 centerInParcels = world.CenterInParcels;
+            MinParcel = centerInParcels - (sizeInParcels / 2);
+            MaxParcel = MinParcel + (sizeInParcels - 1); // last parcel is inclusive
+
+            SizeInUnits = sizeInParcels * parcelSize;
+            MinInUnits = MinParcel * parcelSize;
+            MaxInUnits = MinInUnits + SizeInUnits;
+
+            ChunkModels = new List<ChunkModel>();
+
+            CalculateChunkSizeAndCount();
+            GenerateChunkModels(world.OwnedParcels);
+        }
+
+        private void CalculateChunkSizeAndCount()
+        {
+            int maxSideLengthInUnits = Mathf.Max(SizeInUnits.x, SizeInUnits.y);
+
+            // Determine the number of chunks needed along one side to cover the largest dimension of the terrain
+            sizeInChunks = Mathf.Max(MIN_CHUNKS_PER_SIDE, Mathf.CeilToInt((float)maxSideLengthInUnits / MAX_CHUNK_SIZE));
+
+            ChunkSizeInUnits = Mathf.ClosestPowerOfTwo(maxSideLengthInUnits / sizeInChunks);
+            ChunkSizeInUnits = Mathf.Min(ChunkSizeInUnits, MAX_CHUNK_SIZE); // Ensure it doesn't exceed the max size
+            chunkSizeInParcels = ChunkSizeInUnits / parcelSize;
+
+            if (maxSideLengthInUnits > ChunkSizeInUnits * sizeInChunks)
+                sizeInChunks = Mathf.CeilToInt((float)maxSideLengthInUnits / ChunkSizeInUnits);
+        }
+
+        private void GenerateChunkModels(NativeParallelHashSet<int2> worldOwnedParcels)
+        {
+            ChunkModels = new List<ChunkModel>(sizeInChunks * sizeInChunks);
```

Comment:
It can be moved to the `ctor`, right? There is already a redundant default initialization for no reason?